### DISCUSSION TITLE
[Bug]: SQLSTATE[HY000] [2002] No such file or directory

### DISCUSSION THREAD

**Discussion by myrdrks** [pimcore/pimcore#14255] [Q&A / Support] [answered]
### Expected behavior

Container should start

### Actual behavior

When I follow the steps from xxx I end up after `docker-compose up`.
The console logs:

supervisord_1  | messenger:consume [-l|--limit LIMIT] [-f|--failure-limit FAILURE-LIMIT] [-m|--memory-limit MEMORY-LIMIT] [-t|--time-limit TIME-LIMIT] [--sleep SLEEP] [-b|--bus BUS] [--queues QUEUES] [--no-reset] [--] [<receivers>...]
supervisord_1  |
supervisord_1  | 2023-02-01 16:22:05,891 INFO exited: messenger-consume_00 (exit status 1; not expected)
supervisord_1  | 2023-02-01 16:22:06,901 INFO spawned: 'messenger-consume_00' with pid 5786
supervisord_1  | 2023-02-01 16:22:07,903 INFO success: messenger-consume_00 entered RUNNING state, process has stayed up for > than 0 seconds (startsecs)
supervisord_1  |
supervisord_1  | In AbstractMySQLDriver.php line 112:
supervisord_1  |
supervisord_1  |   An exception occurred in driver: SQLSTATE[HY000] [2002] No such file or dir
supervisord_1  |   ectory
supervisord_1  |
supervisord_1  |
supervisord_1  | In Exception.php line 18:
supervisord_1  |
supervisord_1  |   SQLSTATE[HY000] [2002] No such file or directory
supervisord_1  |
supervisord_1  |
supervisord_1  | In PDOConnection.php line 40:
supervisord_1  |
supervisord_1  |   SQLSTATE[HY000] [2002] No such file or directory

### Steps to reproduce

1. docker run -u `id -u`:`id -g` --rm -v `pwd`:/var/www/html pimcore/pimcore:php8.1-latest composer create-project pimcore/skeleton pimcore
2. cd pimcore
3. echo `id -u`:`id -g` and put in the docker-compose.yaml
4. docker-compose up

I thought I could ignore this and could continue but than I get this after
`docker-compose exec php vendor/bin/pimcore-install --mysql-host-socket=db --mysql-username=pimcore --mysql-password=pimcore --mysql-database=pimcore`

php: symbol lookup error: /usr/local/lib/libMagickCore-7.Q16HDRI.so.10: undefined symbol: heif_init

**Comment by jdreesen** [comment]
I assume you are using an Apple laptop with ARM processor. Please look here: pimcore/docker#121

**Comment by myrdrks** [comment, marked as answer]
In docker-compose.yaml I changed
```
    php:
        image: pimcore/pimcore:php8.1-debug-latest
```
Version to `php8.1-latest`
Now it works

**Comment by dvesh3** [comment]
Moved to discussions so that others can find info about this particular issue. thanks guys 👍

**Reply by KDGundermann** [reply]
Having the same problem here on a fresh Ubuntu 22.04.03 LTS + Docker 27.3.1  installation following https://pimcore.com/docs/platform/2023.3/Pimcore/Getting_Started/Installation/Docker_Based_Installation

```bash
supervisord-1  |
supervisord-1  | In ExceptionConverter.php line 101:
supervisord-1  |
supervisord-1  |   An exception occurred in the driver: SQLSTATE[HY000] [2002] No such file or
supervisord-1  |    directory
supervisord-1  |
supervisord-1  |
supervisord-1  | In Exception.php line 28:
supervisord-1  |
supervisord-1  |   SQLSTATE[HY000] [2002] No such file or directory
supervisord-1  |
supervisord-1  |
supervisord-1  | In Driver.php line 33:
supervisord-1  |
supervisord-1  |   SQLSTATE[HY000] [2002] No such file or directory
supervisord-1  |
supervisord-1  |
supervisord-1  | messenger:consume [-l|--limit LIMIT] [-f|--failure-limit FAILURE-LIMIT] [-m|--memory-limit MEMORY-LIMIT] [-t|--time-limit TIME-LIMIT] [--sleep SLEEP] [-b|--bus BUS] [--queues QUEUES] [--no-reset] [--] [<receivers>...]
supervisord-1  |
```

changing version to `php8.3-latest`did NOT help

**Reply by kingjia90** [reply]
@KDGundermann are you trying to install the Demo, right? looks related to https://github.com/pimcore/demo/issues/591 rather than the original issue here

**Reply by KDGundermann** [reply]
No, I am installing pimcore/skeleton and in Step 3. after `docker compose up` I saw those error messages.

Solution: Just proceed with Step 4 : `docker compose exec php vendor/bin/pimcore-install .... ` and PimCore is running
